extensions/obviousAlexC/3DMath.js

```diff
@@ -0,0 +1,946 @@
+        color1: "#ADC213",
+        color2: "#A0B312",
+        color3: "#697700",
+      };
+    }
+    newV3({ x, y, z }) {
+      return JSON.stringify([
+        Scratch.Cast.toNumber(x) || 0,
+        Scratch.Cast.toNumber(y) || 0,
+        Scratch.Cast.toNumber(z) || 0,
+      ]);
+    }
+    newV3fromValue({ value }) {
+      if (typeof value == "number") {
+        return JSON.stringify([value, value, value]);
+      }
+      return JSON.stringify([0, 0, 0]);
+    }
+    getAxisOfV3({ axis, vector }) {
+      axis = Scratch.Cast.toNumber(axis);
+      vector = JSON.parse(vector);
```

Comment:
you need to handle JSON.parse errors everywhere

easiest might be to make a JSON.parse wrapper that returns `null` if there's an error instead of throwing it so then you can just do `if (!vector1 || !vector2) { return null; }` everywhere